addons/mail/static/src/model/model_manager.js

```diff
@@ -250,6 +250,41 @@ export class ModelManager {
         return res;
     }
 
+    /**
+     * Removes a listener, with the same object reference as given to `startListening`.
+     * Removing the listener effectively makes its `onChange` function no longer
+     * called.
+     *
+     * @param {Object} listener
+     */
+    removeListener(listener) {
+        this._disableListener(listener);
+    }
+
+    /**
+     * Starts a listener. All records/fields access made between `startListening`
+     * and `stopListening` will be saved. When their value later changes,
+     * `listener.onChange` will be called.
+     *
+     * @param {Object} listener
+     * @param {function} listener.onChange function to call when the listened
+     *  records/fields have changed.
+     */
+    startListening(listener) {
+        this._disableListener(listener);
+        this._listeners.add(listener);
+        this._localIdsUsedByListener.set(listener, new Set());
+    }
+
+    /**
+     * Stops a listener, with the same object reference as given to `startListening`.
+     *
+     * @param {*} listener
```

Comment:
I would have liked a class for listener. There is an expected shape for such listeners (`onChange` function, `isPartOfUpdateCycle` boolean, etc.).

Reply:
I'm surprised to see you ask for a class, but I'm ok to add it, I just did the minimal working here (params not really documented etc).

Reply:
The problem with creating objects on the fly in many places and relying on their shape sometime is problematic.

Class helps defining shape of such objects and documenting the attributes.

Personally I have issues with classes when defining methods and slightly annoyed by the poor tools at combining behaviors to objects with class abstraction... But in this case, a class is good because you don't have these problems.